### DISCUSSION TITLE
Flask 10.21.1 has been released! Here's a guide to all the changes

### DISCUSSION THREAD

**Discussion by Montoya** [MetaMask/snaps#897] [Announcements]
As of today, Flask 10.21.1 has been rolled out on Google Chrome and Mozilla Firefox. Here's all the changes: 

1. The Transaction Insights API is now live, [learn how to use it here](https://docs.metamask.io/guide/snaps-exports.html#ontransaction). You can also follow [this workshop](https://hackmd.io/@rekmarks/devcon-vi) by @rekmarks and skip the part about downloading a custom build of Flask.
2. The deprecated `snap_getBip44Entropy_*` method has been removed. Make sure to use [snap_getBip44Entropy](https://docs.metamask.io/guide/snaps-rpc-api.html#snap-getbip44entropy) or [snap_getBip32Entropy](https://docs.metamask.io/guide/snaps-rpc-api.html#snap-getbip32entropy) instead. 
3. A new method [snap_getBip32PublicKey](https://docs.metamask.io/guide/snaps-rpc-api.html#snap-getbip32publickey) is now available. 
4. Lots of bugs have been fixed, including an issue when installing snaps that request the `eth_accounts` permission.

Questions? Issues? Let us know here!

**Comment by marcocastignoli** [comment]
What are the differences between this version and the one available during devcon? The tab in the tx view of my snap is now not visible, no errors on console. I just updated flask to this version, no edit in the code.

Later I tried upgrading the following packages in the snap, but still, no errors in console, and no snap tab in tx
* @metamask/snap-types
* @metamask/snaps-cli

EDIT: in the console I see this warning:
> Error with Permissions-Policy header: Origin trial controlled feature not enabled: 'interest-cohort'.

**Reply by Mrtenz** [reply]
Is your Snap open source, so I can try and reproduce the issue?

**Reply by marcocastignoli** [reply]
ofc! thanks: https://github.com/sourcifyeth/sourcify-metamask-snap

**Reply by marcocastignoli** [reply]
you need to switch on develop!

**Reply by Mrtenz** [reply]
A change we made to allow transaction insights for non-contract interactions (MetaMask/metamask-extension#16093) wasn't included in Flask 10.21 unfortunately. Transaction insights currently only work for contract interactions. It will be included in 10.22 though!

**Reply by marcocastignoli** [reply]
Can you elaborate more? I'm actually interacting with a contract in my example.

**Reply by FrederikBolding** [reply]
@marcocastignoli Are you sure your snap is installing correctly? We lack good feedback right now to indicate when errors occur on install

**Reply by Mrtenz** [reply]
In this case we determine if it's a contract interaction by looking at the data. The test transaction in your Snap doesn't have any data, so it's treated as a regular ETH transfer, rather than a contract interaction.

**Reply by marcocastignoli** [reply]
> @marcocastignoli Are you sure your snap is installing correctly? We lack good feedback right now to indicate when errors occur on install

@FrederikBolding, It shows me the permissions request, and I approve.

**Reply by marcocastignoli** [reply]
@Mrtenz 

> The test transaction in your Snap doesn't have any data, so it's treated as a regular ETH transfer, rather than a contract interaction.

this is my test interaction
```ts
const USDTContract = new ethers.Contract(
    '0xD4B081C226Bc8aBdaf111DEf54c09E779ad29428',
    abi,
    signer,
);

await USDTContract.multiplyBy(3);
```

**Reply by marcocastignoli** [reply]
For some reason the code I cited above is read by metamask as a simple transfer, while the calldata is
```
0xcea299370000000000000000000000000000000000000000000000000000000000000003
```

**Reply by marcocastignoli** [reply]
Ok now I understood. The address I was trying to call is not a contract on mainnet. My mistake :D I can remove this thread, it can confuse people :laughing:

**Reply by Mrtenz** [reply]
No need to remove it, it might help other people running into the same issue.
